source/common/common/utility.h

```diff
@@ -132,16 +134,74 @@ class StringUtil {
+   * Trim trailing whitespaces from a string view.
+   * @param source supplies the string view to be trimmed.
+   * @return trimmed string view.
    */
-  static size_t strlcpy(char* dst, const char* src, size_t size);
+  static absl::string_view rtrim(absl::string_view source);
+
+  /**
+   * Trim leading and trailing whitespaces from a string view.
+   * @param source supplies the string view to be trimmed.
+   * @return trimmed string view.
+   */
+  static absl::string_view trim(absl::string_view source);
 
   /**
+   * Look up for an exactly token in a delimiter-separated string view.
+   * @param source supplies the delimiter-separated string view.
+   * @param multi-delimiter supplies chars used to split the delimiter-separated string view.
+   * @param token supplies the lookup string view.
+   * @param trim_whitespace remove leading and trailing whitespaces from each of the split
+   * string views; default = true.
+   * @return true if found and false otherwise.
+   *
+   * E.g,
+   *
+   * find("A=5; b", "=;", "5")   . true
+   * find("A=5; b", "=;", "A=5") . false
+   * find("A=5; b", "=;", "A")   . true
+   * find("A=5; b", "=;", "b")   . true
+   * find("", ".", "")           . true (spliting "" with "." returns "" which contains "")
+   * find("", ".", "a")          . false (spliting "" with "." returns "" which doesn't contain "a")
+   */
+  static bool find(absl::string_view source, absl::string_view delimiters, absl::string_view token,
+                   bool trim_whitespace = true);
+
+  /**
+   * Crop characters from a string view starting at the first character of the matched
+   * delimiter string view until the end of the source string view.
+   * @param source supplies the string view to be processed.
+   * @param delimiter supplies the string view that delimits the starting point for deletion.
+   * @param trim_whitespace remove leading and trailing whitespaces from each of the split
+   * string views; default = true.
+   * @return sub-string of the string view if any.
+   */
+  static absl::string_view cropRight(absl::string_view source, absl::string_view delimiters,
+                                     bool trim_whitespace = true);
+
+  /**
+   * Split a delimiter-separated string view.
+   * @param source supplies the delimiter-separated string view.
+   * @param multi-delimiter supplies chars used to split the delimiter-separated string view.
+   * @param keep_empty_string result contains empty strings if the string starts or ends with
+   * 'split', or if instances of 'split' are adjacent; default = false.
+   * @return true if found and false otherwise.
+   */
+  static std::vector<absl::string_view> splitToken(absl::string_view source,
+                                                   absl::string_view delimiters,
+                                                   bool keep_empty_string = false);
+
+  /**
+   * TODO(gsagula): remove this when call-sites have been replaced by splitToken().
```

Comment:
Since you are already doing changes to other code sights, is it possible to just delete this in this PR? I doubt there are that many callers and I'm guessing they are all trivial to convert. I've been meaning to do this also.

Reply:
I've already started looking at places where split can be replaced. It looks like that `common/network/utility` is a great candidate for this changes, but would be nice to also convert some of the functions to fully make use of string view. I would be happy to do that in another small PR as soon as I finish the filter, which I think is almost done, if that's ok.

Reply:
Sure, SGTM. Thanks!